lib/rules/no-loss-of-precision.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * @fileoverview Rule to flag numbers that will lose significant figure precision at runtime
+ * @author Jacob Moore
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        type: "problem",
+
+        docs: {
+            description: "disallow literal numbers that lose precision",
+            category: "Possible Errors",
+            recommended: true,
+            url: "https://eslint.org/docs/rules/no-loss-of-precision"
+        },
+        schema: [],
+        messages: {
+            noLossOfPrecision: "This number will lose precision when stored as a Number type."
```

Comment:
```suggestion
            noLossOfPrecision: "This number literal will lose precision at runtime."
```

IMO, "when stored as a Number type" might be a bit redundant and might not do a good job of explaining the issue to someone who's unfamiliar with the rule.

As an optional enhancement, we could include the rounded value in the error message to make the effect of the precision loss clearer.